### ISSUE TITLE
Pi node availability time not updating and no bonus rewards 

### ISSUE THREAD

**Issue by sureshjkb** [pi-node/instructions#286]
The required parameters of my  test node is  ok. But nit getting any bonus reward from so many days . Neither the availability time is updating

**Comment by NeyMR** [comment]
⚠️Pi Node Availability Tuning Formula⚠️
Node formula calculates the Availability (Uptime), Ports Open, and CPU performance factors are all added together and then multiplied by the Node Uptime from the previous day. If the Node was inactive, its uptime is 0 and 0 * Anything is 0. There is no passive reward for past activity alone. The same applies to the App usage reward, if you don't use apps the day before then there is no reward on the current day, so your nodes need to be up for tuning to calculate the bonus for node availability…  All this information can be found: pinetwork.atlassian.net/servicedesk/customer/portal/1/article/956792833?src=508365090

please have patience and let the tuning process conclude all bonuses will be back to normal and calculated, please always check Dev channels 🔝👆Pinned Messages 👆Mods FAQ👆🔝 it will have as well all the information just provided and more. thanks for your understanding, we're all in this together 😁👍

**Comment by Macromart** [comment]
Mr NeyMR  Your reasoning is NOT the issue. Its caused by the programmer not keeping the node software updated. It's that simple!!

**Comment by cheme75** [comment]
I agree, not one single update in the 2 yrs since last issued is not how responsible developers behave. There’s no way this software is release candidate ready, plus users have pointed out ongoing security risks that could be patched, if the developers cared. I also agree there are serious flaws that ignore nodes at what seems randomly and intermittently. I can see my node is clearly running with significant incoming and outgoing connections and supporting other nodes, yet I can go a week with zero node bonus. What’s worse is that the developers do not care! We expend money and effort to support the network, our bonus should only drop to zero if the node goes down or otherwise is unreachable.

**Comment by t4ndu** [comment]
when they solve the problem with the bonus.
when comes a dashboard for analytics?
when comes an allert when the node sopp sync.

**Comment by Macromart** [comment]
Couldn't agree more. It's all like a movie set facade. Looks good from the
front, but open the front door and it's a vacant block of land. No one
answering complaints about crazy over zealous chat hosts giving out abuse
strikes for asking questions. No support, no one answering anything. I've
tried to get answers about node security issues and no one there. It's like
a big hoax. I took down my node yesterday. Had enough. It's all too silly
for words.

I suggest everyone else does the same thing. It's being run too badly to
trust......bye for good.  Having digital currencies in private hands is a
huge risk. This pi crowd in my opinion are so vindictive, they cut me from
KYC next for posting the truth......I don't care. All leaves a very sour
taste......

**Comment by cheme75** [comment]
> Couldn't agree more. It's all like a movie set facade. Looks good from the front, but open the front door and it's a vacant block of land. No one answering complaints about crazy over zealous chat hosts giving out abuse strikes for asking questions. No support, no one answering anything. I've tried to get answers about node security issues and no one there. It's like a big hoax. I took down my node yesterday. Had enough. It's all too silly for words. I suggest everyone else does the same thing. It's being run too badly to trust......bye for good. Having digital currencies in private hands is a huge risk. This pi crowd in my opinion are so vindictive, they cut me from KYC next for posting the truth......I don't care. All leaves a very sour taste......
> […](#)

Yup, I agree. I’m only running a node since I’m running another container on same machine. But that other app is well maintained and frequently updated. I asked the dev about some issues getting it running, I had answers same or next day, I asked about adding a feature, it was implemented next day! He is frequently upgrading and adding functionality. This app/system gets nothing. I’ve tried to submit support tickets for app glitches and the node situation or the node security issues that are being ignored, I tried posting brainstorms to add a private node operator info chat, suggestions to make all chats at least threaded to be actually useful, many other suggestions - zero replies, zero fixes, zero updates, zero improvements. I’ll let the node continue as long as the other container is working ok, although I’d think the other would run even better on Linux due to all the heavy overhead of windows. So if I move that other container to Linux that will force me to eliminate this node when I reformat drive for Linux. I’m not expecting a Linux option for the node. I’m not expecting any improvement. I think the developers are too obscure and not being at all transparent about what’s going on behind the scenes. If we are all in this together, then we deserve some informative updates, not just marketing spin videos that just offer hype and leave out the facts. Here’s one of the better analysis of pi that I’ve read: https://tradecrypto.com/academy/altcoin-academy/pi-network-full-review/

**Comment by NeyMR** [comment]
> > Couldn't agree more. It's all like a movie set facade. Looks good from the front, but open the front door and it's a vacant block of land. No one answering complaints about crazy over zealous chat hosts giving out abuse strikes for asking questions. No support, no one answering anything. I've tried to get answers about node security issues and no one there. It's like a big hoax. I took down my node yesterday. Had enough. It's all too silly for words. I suggest everyone else does the same thing. It's being run too badly to trust......bye for good. Having digital currencies in private hands is a huge risk. This pi crowd in my opinion are so vindictive, they cut me from KYC next for posting the truth......I don't care. All leaves a very sour taste......
> > […](#)
> 
> Yup, I agree. I’m only running a node since I’m running another container on same machine. But that other app is well maintained and frequently updated. I asked the dev about some issues getting it running, I had answers same or next day, I asked about adding a feature, it was implemented next day! He is frequently upgrading and adding functionality. This app/system gets nothing. I’ve tried to submit support tickets for app glitches and the node situation or the node security issues that are being ignored, I tried posting brainstorms to add a private node operator info chat, suggestions to make all chats at least threaded to be actually useful, many other suggestions - zero replies, zero fixes, zero updates, zero improvements. I’ll let the node continue as long as the other container is working ok, although I’d think the other would run even better on Linux due to all the heavy overhead of windows. So if I move that other container to Linux that will force me to eliminate this node when I reformat drive for Linux. I’m not expecting a Linux option for the node. I’m not expecting any improvement. I think the developers are too obscure and not being at all transparent about what’s going on behind the scenes. If we are all in this together, then we deserve some informative updates, not just marketing spin videos that just offer hype and leave out the facts. Here’s one of the better analysis of pi that I’ve read: https://tradecrypto.com/academy/altcoin-academy/pi-network-full-review/

All this anger from not getting KYC when many are waiting, We have family members still waiting patiently like millions of others, just have some patience and stop listening to rumours as the project is moving forward strong, thanks for your understanding.

**Comment by Macromart** [comment]
What on earth has node issues got to do with KYC. You need to read the
story before blurting Pi "back slaps" dude.

**Comment by cheme75** [comment]
> What on earth has node issues got to do with KYC. You need to read the story before blurting Pi "back slaps" dude.

Ya, agree, he is no help. Why bother replying to issues if you can’t offer any actual assistance? I ignore this guy. Either it’s a English translation problem or he’s simply a pi acolyte hooked on the koolaid. But indirectly, there is some wording that nodes will be selected for main net only after passing kyc. However, that does not explain the total lack of attention to those who have exposed glitches, bugs and security risks or simply want some answers about their plans for Linux or other platforms. Plus the draconian behavior of the chat mods - I rarely use the chat since it’s a farce - seems like nothing ever gets a reply from anyone knowledgeable and most mod replies are useless. They seriously need to engage here on this forum!  I’m pretty close to shutting down my node. I use docker for a platform for some cams, I’d like to switch to Linux for it but this node is stuck with windows, but windows has so much useless overheads, machine bogs down too often. Linux is so much cleaner, it’s a pity they’ve ignored it.

**Comment by Macromart** [comment]
I agree with you. I'm in Australia. I saw the nodes counted per country
recently. China has the most nodes 41,000 but Australia has only 480 or so.
That is incredibly low!!!  I get why the Chinese are into it in a big way,
but for Australia with over 27 million people, that is a very small pool of
people. I wonder how many people there are like me who just got tired of
talking to the Pi "brick wall" and turned off their node & walked away? I
have NEVER received a reply to any of my question or emails. THAT IS VERY
TROUBLING.

**Comment by cheme75** [comment]
I’m in the US. Got no idea on how many here, probably quite a bit but certainly less than China. I joined up since my son asked and my research found nothing definitive about it being a scam, although we both have our doubts. I’m retired so got the time. Before I broke down my old desktop to sell the parts, I figured why not run a node and get the bonus since my team of 35 appears to have dwindled to no more than 2-5 regulars still hanging in. It’s a 2nd gen i5, but struggles with the node plus the HD is only 300gb so limited. If they had a Linux version it would a big improvement. I’m running lubuntu on an even older netbook for homebridge and scrypted which run very well. I’d switch pc to Linux in a heartbeat. I’d even invest in a rpi if they were to offer it. But I refuse to invest any more than the electricity and time to run the old pc until I see some substantive support. 

Fwiw, I’ve only been mining for a bit over yr and a half now so it’s been at relatively low mining rates and with 80% of my team gone, I’ll probably lose significant pi if I ever get the kyc invite. Assuming pi survives, there is still a long road before it would be listed on an exchange which is what is needed to “get rich”but only if it were worth $1000/pi or more. What concerns me, it is not scarce and it is not difficult to mine, so has none of the critical attributes that make other crypto valuable, ie: bitcoin. Eth is next I believe and has no cap and is less than 1/10 bitcoin and consider how long and how costly it is to mine eth. To think pi would be on a par with eth is delusional. 

But debating pi value is simply conjecture. The whole crypto value proposition may or may not ever become a common currency, only time will tell. Meanwhile, I completely agree with you. I’ve submitted my share of support emails and nothing ever gets answered. To believe Dr @kokkalis and his PhD partners are serious about this is hard to swallow at this point. Considering their pedigree, this could be just a giant social experiment to prove how easy it can be to fool people on a global scale, then they can reap the rewards of the book and movie 😉

Troubling is certainly a good word for it: KYC is slow and I’m still on the fence about the 3rd party security. Nodes are the life line of the network and there has been zero support, zero feedback, zero upgrades, zero concern about exposed security risks in the libraries they continue to deploy, zero transparency, a few anecdotal platitudes about the ongoing efforts, but nothing substantive. This is not how a responsible organization behaves.

**Comment by steeveee100** [comment]
While trying to sort out the node lack-of-reward issues I had a look at the Reward page. That's the one that starts with Total mining rate!

At the bottom it says "Node Bonus" and, in red, "Tuning in progress. This value may change"
Below that is a button marked INCREASE. 

When you hit that it asks for a sign-in code. Has anyone done this? Mine is blank and I don't remember where to find this code? Can anyone help?

**Comment by cheme75** [comment]
> While trying to sort out the node lack-of-reward issues I had a look at the Reward page. That's the one that starts with Total mining rate!
> 
> At the bottom it says "Node Bonus" and, in red, "Tuning in progress. This value may change" Below that is a button marked INCREASE.
> 
> When you hit that it asks for a sign-in code. Has anyone done this? Mine is blank and I don't remember where to find this code? Can anyone help?

On initial setup, you get the code from the pc app. Enter it on mobile app to authenticate the node login. I never wrote it down either. It appears to be a one time code and it seems they’ve never added the ability to view the node from the mobile app. When I decided to move HD to bring my old desktop back to life, I had to re-login to pc app and that’s the only time I ever got a code again on the pc, for the login of the node by authenticating since it was a change of cpu, motherboard, etc. I didn’t see any use to write it down since it seemed like a one time use, not a static code. But I may be wrong. Possibly if you can sign out of pc app, it may generate the code to sign back in. I haven’t tried that.

**Comment by ROSEIBYE** [comment]
I want to use my national ID new surname . It's different from the surname I used in registration. How do I go about it

**Comment by Tbombs** [comment]
> > What on earth has node issues got to do with KYC. You need to read the story before blurting Pi "back slaps" dude.
> 
> Ya, agree, he is no help. Why bother replying to issues if you can’t offer any actual assistance? I ignore this guy. Either it’s a English translation problem or he’s simply a pi acolyte hooked on the koolaid. But indirectly, there is some wording that nodes will be selected for main net only after passing kyc. However, that does not explain the total lack of attention to those who have exposed glitches, bugs and security risks or simply want some answers about their plans for Linux or other platforms. Plus the draconian behavior of the chat mods - I rarely use the chat since it’s a farce - seems like nothing ever gets a reply from anyone knowledgeable and most mod replies are useless. They seriously need to engage here on this forum! I’m pretty close to shutting down my node. I use docker for a platform for some cams, I’d like to switch to Linux for it but this node is stuck with windows, but windows has so much useless overheads, machine bogs down too often. Linux is so much cleaner, it’s a pity they’ve ignored it.

yeah the chat seems to be somewhat useless, and any legit problems or issues that can't be solved by "reading the whitepaper" or read the "mods faq" its like the every answer for someone is just read this or read that.  

I will say thought, the chat channels can be flooded with lots of basic questions and issues and either the people are just not great with technology or reading lol, or they are just kind of dumb tbh.  I.e. Pioneer complaining about KYC, then find out they used their MOTHERS passport lmao ofc it failed ya dingus. 

But because of these common issues of people not following directions or not reading whitepaper, then when those of us actually have non typical problems we are treated like the dumb people who can't read directions etc.  so i see both sides of it but most of the time i feel like chat isn't worth it.  Github/reddit are much better. also found a telegram group for pi node if anyone is int i will link.  

[PiNodeChat "Pi Node help and news etc, 0 KYC blah blah blah 😁👍
" lmao](https://t.me/PiNodechat)

**Comment by Tbombs** [comment]
> > What on earth has node issues got to do with KYC. You need to read the story before blurting Pi "back slaps" dude.
> 
> Ya, agree, he is no help. Why bother replying to issues if you can’t offer any actual assistance? I ignore this guy. Either it’s a English translation problem or he’s simply a pi acolyte hooked on the koolaid. But indirectly, there is some wording that nodes will be selected for main net only after passing kyc. However, that does not explain the total lack of attention to those who have exposed glitches, bugs and security risks or simply want some answers about their plans for Linux or other platforms. Plus the draconian behavior of the chat mods - I rarely use the chat since it’s a farce - seems like nothing ever gets a reply from anyone knowledgeable and most mod replies are useless. They seriously need to engage here on this forum! I’m pretty close to shutting down my node. I use docker for a platform for some cams, I’d like to switch to Linux for it but this node is stuck with windows, but windows has so much useless overheads, machine bogs down too often. Linux is so much cleaner, it’s a pity they’ve ignored it.

hey just FYI docker on windows actually does use a linux distribution as part of its windows operation.  yes its not like booting up another OS but it does show up in file explorer and it utilized to make docker run more efficiently and effectively.  

If you are really set on running a Linux OS, i would recommend either using a seperate hard drive, installing a Virtual Machine (VM) to run it in a window inside your windows OS, or you could partition your Hard drive to have the LINUX dist as a dual boot option when doing POST.  VM is probably the safest easiest and then if you make any mistakes etc then big whoop it was a VM :)

**Comment by Tbombs** [comment]
ok so after a quick google search apparently docker already does run linux as a VM lol its called WSL2

"1. Docker Desktop with WSL 2 (Recommended):
This is the recommended way to run Docker on Windows.
WSL 2 (Windows Subsystem for Linux 2) provides a full Linux kernel within Windows, allowing Linux containers to run natively without emulation.

Docker Desktop leverages WSL 2 to provide a seamless experience for building and running both Linux and Windows containers.

2. Docker Desktop with Hyper-V (Older versions):
In older versions of Docker Desktop and Windows, Linux containers ran inside a lightweight Linux VM (Moby Linux VM) powered by Hyper-V.
This method is still supported but not as performant as using WSL 2.
In both cases:

You interact with Docker using the same Docker commands and tools regardless of whether you're running Linux or Windows containers.
The Docker engine running within WSL 2 or the Hyper-V VM manages the containers.

Docker Desktop creates a virtual machine. This is why you can run Linux containers inside that VM, but you can't use it for running Windows containers and Linux containers at the same time.Jan 26, 2023"

ANYWAYS good luck with pi node and getting your bonus. i haven't figured out how to quickly resolve a 0 bonus because if we don't know there is a problem until the next mining session its hard to fix it lol.  and since bonus is multiplied by a few things if one is 0 then ofc ur bonus will be zero just a thought.
